### PR TITLE
Fix MethodHandle exception wrapping

### PR DESCRIPTION
**Pull request by theresa-m**
Signed-off-by: Theresa Mammarella <Theresa.T.Mammarella@ibm.com>

### CODE REVIEW

**Comment by DanHeidinga** [comment]
@theresa-m I'm not clear on why this is required.  Is there a spec reference that requires this?  Is this behaviour consistent between JDK8 / JDK11 / JDK12 & JDK13?

File: jcl/src/java.base/share/classes/java/lang/invoke/PrimitiveHandle.java [hunk 1]
```diff
@@ -192,7 +192,9 @@ public MethodHandle asVarargsCollector(Class<?> arrayParameter) throws IllegalAr
 			String signature = oldType.toMethodDescriptorString();
 			return lookupMethod(referenceClass, name, signature, kind, specialToken);
 		} catch (NoSuchMethodError e) {
-			throw new NoSuchMethodException(e.getMessage());
+			NoSuchMethodException eWrapper = new NoSuchMethodException();
+			eWrapper.initCause(e);
+			throw eWrapper;
```
Comment (DanHeidinga) [inline jcl/src/java.base/share/classes/java/lang/invoke/PrimitiveHandle.java]: I'm OK with this change as the NSME already exists

File: jcl/src/java.base/share/classes/java/lang/invoke/MethodHandles.java [hunk 2]
```diff
@@ -525,7 +525,11 @@ private void checkAccess(Class<?> definingClass, Class<?> referenceClass, String
 			}
 			/*[MSG "K0587", "'{0}' no access to: '{1}'"]*/
 			String errorMessage = com.ibm.oti.util.Msg.getString("K0587", this.toString(), definingClass.getName() + extraInfo);  //$NON-NLS-1$
-			throw new IllegalAccessException(errorMessage);
+
+			/* Error should be IllegalAccessException caused by IllegalAccessError */
+			IllegalAccessException iae = new IllegalAccessException();
+			iae.initCause(new IllegalAccessError(errorMessage));
+			throw iae;
```
Comment (DanHeidinga) [inline jcl/src/java.base/share/classes/java/lang/invoke/MethodHandles.java]: This feels really hacky and we shouldn't do this.  Any test requiring this is bogus.

File: jcl/src/java.base/share/classes/java/lang/invoke/PrimitiveHandle.java (hunk 1 above)
Comment (theresa-m) [reply jcl/src/java.base/share/classes/java/lang/invoke/PrimitiveHandle.java]: updated to exclude the changes to MethodHandles.Lookup.findStatic.

**Comment by DanHeidinga** [comment]
Jenkins test sanity win jdk8

**Pull request merged**